coderadar-test/src/test/java/io/reflectoring/coderadar/rest/useradministration/permissions/RemoveUserFromProjectControllerIntegrationTest.java

```diff
@@ -0,0 +1,87 @@
+package io.reflectoring.coderadar.rest.useradministration.permissions;
+
+import static io.reflectoring.coderadar.rest.JsonHelper.fromJson;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import io.reflectoring.coderadar.graph.projectadministration.domain.ProjectEntity;
+import io.reflectoring.coderadar.graph.projectadministration.project.repository.ProjectRepository;
+import io.reflectoring.coderadar.graph.useradministration.domain.UserEntity;
+import io.reflectoring.coderadar.graph.useradministration.repository.UserRepository;
+import io.reflectoring.coderadar.rest.ControllerTestTemplate;
+import io.reflectoring.coderadar.rest.domain.ErrorMessageResponse;
+import io.reflectoring.coderadar.useradministration.service.security.PasswordUtil;
+import java.util.Collections;
+import java.util.Date;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.web.servlet.MvcResult;
+
+class RemoveUserFromProjectControllerIntegrationTest extends ControllerTestTemplate {
+
+  @Autowired private UserRepository userRepository;
+
+  @Autowired private ProjectRepository projectRepository;
+
+  private ProjectEntity testProject;
+  private UserEntity testUser;
+
+  @BeforeEach
+  void setUp() {
+    testProject = new ProjectEntity();
+    testProject.setVcsUrl("https://valid.url");
+    testProject.setName("project");
+    testProject.setVcsStart(new Date());
+    testProject.setVcsOnline(true);
+    testProject.setVcsPassword("testPassword");
+    testProject.setVcsUsername("testUser");
+    projectRepository.save(testProject);
+
+    testUser = new UserEntity();
+    testUser.setUsername("username");
+    testUser.setPassword(PasswordUtil.hash("password1"));
+    testUser.setProjects(Collections.singletonList(testProject));
+    userRepository.save(testUser);
+  }
+
```

Comment:
What happens if there is no user with this id for this project? What happens if user and project do not exist?

Reply:
I'm not sure what you mean. The cases you describe are tested on lines 77 and 94.

Reply:
The first means the user exists (has id 1) but he is not a user registered for this project. The test case you've used in line 77 is for a user which does not exist on our platform. For the second part, this file only does have 87 lines, at least in the GitHub view. What test is written in line 94?

Reply:
Sorry, the lines I mentioned are for the `SetUserRoleForProjectIntegrationTest`. The first test checks if  a user assigned to a project gets their role removed successfuly. The other two tests check that 404 is returned when a user/project does not exists. Is there something I'm missing?